Towny/src/main/java/com/palmergames/bukkit/util/ItemLists.java

```diff
@@ -266,106 +240,19 @@ public boolean contains(@NotNull ItemStack itemStack) {
-		private final Set<Predicate<String>> anyMatchPredicates;
-
-		private PredicateList(@NotNull Set<Predicate<String>> predicates, @NotNull Set<Predicate<String>> orPredicates) {
-			this.allMatchPredicates = predicates;
-			this.anyMatchPredicates = orPredicates;
-		}
-		
-		private boolean contains(@NotNull Material material) {
-			return allMatchPredicates.stream().allMatch(predicate -> predicate.test(material.name())) && anyMatchPredicates.stream().anyMatch(predicate -> predicate.test(material.name()));
-		}
-		
-		private static PredicateListBuilder builder() {
-			return new PredicateListBuilder().notStartsWith("LEGACY_"); // Excludes any legacy materials making it into our results.
-		}
-	}
-
-	private static class PredicateListBuilder {
-		private final Set<Predicate<String>> allMatchPredicates = new HashSet<>();
-		private final Set<Predicate<String>> anyMatchPredicates = new HashSet<>();
-		private @Nullable EnumSet<Material> exceptions;
-
-		public ItemLists build() {
-			PredicateList predicateList = new PredicateList(allMatchPredicates, anyMatchPredicates);
-			EnumSet<Material> matches = EnumSet.noneOf(Material.class);
-
-			for (Material material : Material.values())
-				if (predicateList.contains(material))
-					matches.add(material);
-			
-			if (exceptions != null)
-				matches.addAll(exceptions);
-			
-			return new ItemLists(matches);
-		}
-
-		public PredicateListBuilder startsWith(String startingWith) {
-			anyMatchPredicates.add((s) -> s.startsWith(startingWith));
-			return this;
-		}
-
-		public PredicateListBuilder endsWith(@NotNull String endingWith) {
-			anyMatchPredicates.add((s) -> s.endsWith(endingWith));
-			return this;
-		}
-
-		public PredicateListBuilder not(@NotNull String name) {
-			allMatchPredicates.add((s) -> !s.equals(name));
-			return this;
-		}
-
-		public PredicateListBuilder notStartsWith(@NotNull String notStartingWith) {
-			allMatchPredicates.add((s) -> !s.startsWith(notStartingWith));
-			return this;
-		}
-
-		@SuppressWarnings("unused")
-		public PredicateListBuilder notEndsWith(@NotNull String notEndingWith) {
-			allMatchPredicates.add((s) -> !s.endsWith(notEndingWith));
-			return this;
-		}
-		
-		@SuppressWarnings("unused")
-		public PredicateListBuilder contains(@NotNull String containing) {
-			allMatchPredicates.add((s) -> s.contains(containing));
-			return this;
-		}
-		
-		@SuppressWarnings("unused")
-		public PredicateListBuilder notContains(@NotNull String notContaining) {
-			allMatchPredicates.add((s) -> !s.contains(notContaining));
-			return this;
-		}
-
-		public PredicateListBuilder add(@NotNull String... names) {
-			if (exceptions == null)
-				exceptions = EnumSet.noneOf(Material.class);
-
-			for (String name : names) {
-				try {
-					exceptions.add(Material.valueOf(name.toUpperCase(Locale.ROOT)));
-				} catch (IllegalArgumentException ignored) {}
-			}
-
-			return this;
-		}
+	public static Builder<Material, ItemLists> newBuilder() {

```

Comment:
Nice to see a public constructor, Goosius and I had just recently had an issue where we couldn't use ItemLists because PredicateLists couldn't be made with other plugins.